src/matrix/lkj.jl

```diff
@@ -0,0 +1,239 @@
+"""
+    LKJ(d, η)
+```julia
+d::Int   dimension
+η::Real  positive shape
+```
+The [LKJ](https://doi.org/10.1016/j.jmva.2009.04.008) distribution is a distribution over
+``d\\times d`` real correlation matrices (positive-definite matrices with ones on the diagonal).
+If ``\\mathbf{R}\\sim \\textrm{LKJ}_{d}(\\eta)``, then its probability density function is
+
+```math
+f(\\mathbf{R};\\eta) = \\left[\\prod_{k=1}^{d-1}\\pi^{\\frac{k}{2}}
+\\frac{\\Gamma\\left(\\eta+\\frac{d-1-k}{2}\\right)}{\\Gamma\\left(\\eta+\\frac{d-1}{2}\\right)}\\right]^{-1}
+|\\mathbf{R}|^{\\eta-1}.
+```
+
+If ``\\eta = 1``, then the LKJ distribution is uniform over
+[the space of correlation matrices](https://www.jstor.org/stable/2684832).
+"""
+struct LKJ{T <: Real, D <: Integer} <: ContinuousMatrixDistribution
+    d::D
+    η::T
+    logc0::T
+end
+
+#  -----------------------------------------------------------------------------
+#  Constructors
+#  -----------------------------------------------------------------------------
+
```

Comment:
You could use a keyword `check_args=true` to avoid checking multiple times when creating the distribution, and `@check_args(LKJ, d > 0 && η > 0)`. The latter is up to your preference, both are fine with me

Reply:
> to avoid checking multiple times when creating the distribution

Not sure what you mean.

Reply:
sorry yes. When creating multiple time the same distribution, or when copying a distribution, it's practical to have a method shortcircuit all the checks

Reply:
Okey-doke. Done.